--- linux.mk	2025-02-11 11:40:43.431939117 +0100
+++ targets/linux.mk	2025-02-11 11:38:31.186995411 +0100
@@ -13,32 +13,15 @@
 
 kernel_version = $(CONFIG_LINUXBOOT_KERNEL_VERSION)
 kernel_dir = build/kernel-$(subst .,_,$(kernel_version))
-kernel_tarball = linux-$(kernel_version).tar
-kernel_mirror  = https://mirrors.edge.kernel.org/pub/linux/kernel
+# kernel_tarball = master.tar
+kernel_mirror  = https://github.com/micgor32/linux.git
 
-ifeq ($(findstring x2.6.,x$(kernel_version)),x2.6.)
-kernel_mirror_path := $(kernel_mirror)/v2.6
-else ifeq ($(findstring x3.,x$(kernel_version)),x3.)
-kernel_mirror_path := $(kernel_mirror)/v3.x
-else ifeq ($(findstring x4.,x$(kernel_version)),x4.)
-kernel_mirror_path := $(kernel_mirror)/v4.x
-else ifeq ($(findstring x5.,x$(kernel_version)),x5.)
-kernel_mirror_path := $(kernel_mirror)/v5.x
-else ifeq ($(findstring x6.,x$(kernel_version)),x6.)
-kernel_mirror_path := $(kernel_mirror)/v6.x
-endif
-
-build/$(kernel_tarball).xz: | build
+build/$(kernel_dir): | build
 	echo "    Test $(kernel_version)"
-	echo "    WWW        $(kernel_mirror_path)/$(kernel_tarball).xz";
-	curl -OLSs --output-dir build "$(kernel_mirror_path)/$(kernel_tarball).xz";
-
-$(kernel_dir): build/$(kernel_tarball).xz
-	echo "    XZ         $(kernel_tarball).xz";
-	mkdir $(kernel_dir);
-	tar xJf build/$(kernel_tarball).xz --strip 1 -C $(kernel_dir);
+	echo "    Git clone $(kernel_mirror)";
+	git clone "$(kernel_mirror)" "$(kernel_dir)";
 
-$(kernel_dir)/.config: $(CONFIG_LINUXBOOT_KERNEL_CONFIGFILE) | $(kernel_dir)
+$(kernel_dir)/.config: $(CONFIG_LINUXBOOT_KERNEL_CONFIGFILE) | build/$(kernel_dir)
 	@echo "    CONFIG     Linux $(kernel_version)"
 	cp $(CONFIG_LINUXBOOT_KERNEL_CONFIGFILE) $(kernel_dir)/.config
 	$(MAKE) -C $(kernel_dir) $(KERNEL_MAKE_FLAGS) olddefconfig
